### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by rakkit**
bug fix, see comments

### CODE REVIEW

File: torchtitan/models/inits.py [hunk 1]
```diff
@@ -22,30 +22,33 @@ def orthogonal_(params, gain: float = 1.0, generator: Optional[torch.Generator]
     if not isinstance(params.data, DTensor):
         return nn.init.orthogonal_(params, gain=gain, generator=generator)
     else:
-        temp_tensor = torch.empty(params.shape)  # full shape
+        temp_tensor = torch.empty(params.shape, device=params.device)  # full shape
```
Comment (rakkit) [inline torchtitan/models/inits.py:25]: we should explicitly move `temp_tensor` on GPU otherwise, it runs qr-decomp on CPU

File: torchtitan/models/inits.py [hunk 2]
```diff
@@ -22,30 +22,33 @@ def orthogonal_(params, gain: float = 1.0, generator: Optional[torch.Generator]
...
         torch.nn.init.orthogonal_(temp_tensor, gain=gain, generator=generator)
 
-        # Create a replicated DTensor
-        replicated = DTensor.from_local(
-            temp_tensor,
-            placements=[Replicate()] * params.device_mesh.ndim,
-            device_mesh=params.device_mesh,
+        params_data = distribute_tensor(
```
Comment (rakkit) [inline torchtitan/models/inits.py:28]: simplify code via `distribute_tensor`

File: torchtitan/optimizers/distributed_scion.py [hunk 3]
```diff
@@ -130,13 +130,26 @@ def step(self, closure=None):
...
                 loss = closure()
-
+        sgd_params = []
         ddp_params = []
         fsdp_params = []
         expert_params = []
 
         for group in self.param_groups:
             for p in group["params"]:
+                param_kwargs = self.groups_info[self.paramters_to_groups[id(p)]][-1]
```
Comment (rakkit) [inline torchtitan/optimizers/distributed_scion.py:155]: for SGD parameters (backend=identical and norm is Row-wise normalized) we don't need to gather them

File: torchtitan/optimizers/distributed_scion.py [hunk 4]
```diff
@@ -161,18 +175,20 @@ def lmo(self, g, eps, norm_factor, zeropower_backend, backend_steps):
         # NB: make sure this function does not modify the grad inplace
         #     since it is also called during the log of gradients
         # g = zeropower_backend(g, steps=backend_steps, eps=eps)
-        if g.ndim == 2:
+        def _lmo_for_2d_tensor(g):
```
Comment (rakkit) [inline torchtitan/optimizers/distributed_scion.py:199]: for 3d weights, we should apply `nomrlize_grad` on each 2d weights respectively

**Comment by rakkit** [comment]
Can merge this PR first `https://github.com/janEbert/torchtitan/pull/13`

then only take the changes in `torchtitan/optimizers/distributed_scion.py` taken.  the API is aligned to PR#13

**Comment by janEbert** [comment]
Merged.
